### PR TITLE
Added tanh activation and a way to switch between different functions

### PR DESCRIPTION
**Pull request by schrummy14**
Also removed function so user does not have to choose both the correct activation function with the correct derivative of the activation function

### CODE REVIEW

**Comment by schrummy14** [comment]
When testing with the xor..... it is a little sensitive to the learning rate......

File: lib/nn.js [hunk 1]
```diff
@@ -54,12 +60,14 @@ class NeuralNetwork {
     this.learning_rate = learning_rate;    
   }
   
-  setActivationFunction(Fun = sigmoid) {
-    this.activation_function = Fun;
-  }
-  
-  setDActivationFunction(dFun = dsigmoid) {
-    this.d_activation_function = dFun;
+  setActivationFunction(Fun = 'sigmoid') {
```
Comment (shiffman) [inline lib/nn.js]: Would you mind using `func` instead of `Fun`?

File: lib/nn.js [hunk 2]
```diff
@@ -9,6 +9,13 @@ function dsigmoid(y) {
   return y * (1 - y);
 }
 
+function tanh(x) {
+	return Math.tanh(x);
+}
+function dtanh(x) {
+	return 1-(tanh(x)*tanh(x));
```
Comment (shiffman) [inline lib/nn.js]: Since the output of a node has already been passed through `tanh()` I believe this would be `(1-y*y)`? (The convention I'm using to note this is `y` instead of `x`. Am I mistaken about this?

File: lib/nn.js [hunk 3]
```diff
@@ -9,6 +9,13 @@ function dsigmoid(y) {
   return y * (1 - y);
 }
 
+function tanh(x) {
+	return Math.tanh(x);
```
Comment (shiffman) [inline lib/nn.js]: Would you mind conforming to the 2 space indentation?

**Comment by schrummy14** [comment]
Made changes and fixed a confilict

**Comment by arisanguinetti** [comment]
this seems to be covered on PR #62...

**Comment by shiffman** [comment]
Yes indeed, apologies @schrummy14 but I just couldn't resist the object-oriented solution.

**Comment by schrummy14** [comment]
That is fine. It will allow for a cleaner solution.
